packages/astro/src/core/build/generate.ts

```diff
@@ -191,7 +191,7 @@ async function generatePath(
 	}
 
 	const ssr = isBuildingToSSR(opts.astroConfig);
-	const url = new URL(origin + pathname);
+	const url = new URL(opts.astroConfig.base + pathname.substring(1), origin);
```

Comment:
I think the answer here is yes, but is it safe to always assume `pathname` will start with a `/` here?

If not, may be a good spot for a regex that only strips the first character if it's a `/`

Reply:
yea, agreed. better safe than sorry